src/cider/tests/nextgen/functional/PrimitiveTypeQueryNextgenTest.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <gtest/gtest.h>
+#include "exec/nextgen/Nextgen.h"
+#include "exec/plan/parser/TypeUtils.h"
+#include "tests/utils/ArrowArrayBuilder.h"
+#include "tests/utils/CiderTestBase.h"
+#include "tests/utils/QueryArrowDataGenerator.h"
+
+#define GEN_PRIMITIVETYPE_BASE_TEST_CLASS(C_TYPE_NAME, TYPE, SUBSTRAIT_TYPE_NAME)  \
```

Comment:
Minor hint: For file name, we can just simplify it to ```PrimitiveTypeTest.cpp```. File path already includes ```nextgen```.

Reply:
`PrimitiveTypeTest.cpp` already exists under cider/tests. It will cause duplication and replacement if we still use this name here. But eventually, we will format the test file names when previous UTs removed.